src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -543,6 +539,7 @@ private async Task RefreshKeyValueCollections(CancellationToken cancellationToke
                     });
                 }
 
+                changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(changeWatcher.CacheExpirationInterval);

```

Comment:
Can we move this line right after the `if (DateTimeOffset.UtcNow < changeWatcher.CacheExpires)`, so it's clear the extension is the first thing we do every time.